pkg/csi/cinder/openstack/openstack.go

```diff
@@ -74,6 +83,7 @@ type BlockStorageOpts struct {
 type Config struct {
 	openstack_provider.Config
 	BlockStorage BlockStorageOpts
+	Metadata     MetadataOpts
```

Comment:
This option is already included in `openstack_provider.Config`.

Reply:
@lingxiankong But metadata opts in cloudprovider config includes additional options as well which is not yet supported by csi cinder, so isnt it better to override?
https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/cloudprovider/providers/openstack/openstack.go#L164

Reply:
maybe later on when we do the refactory, we can consolidate then?

Reply:
> @lingxiankong But metadata opts in cloudprovider config includes additional options as well which is not yet supported by csi cinder, so isnt it better to override?

If ignoring the extra options doesn't break cinder-csi, that would be ok. The config options for those services should be well documented.

Reply:
> maybe later on when we do the refactory, we can consolidate then?

But if we add `Metadata` option here instead of re-using https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/cloudprovider/providers/openstack/openstack.go#L181, it will be hard for us to refactor in the future because we will introduce incompatible config change.

Reply:
I tested the configuration ,with existing code add cloud.conf by adding [metadata] section
and works fine, I guess we didn't introduce incompatible config change?

my env is down now ,but I am sure it's something like

```
[Global]
username=user
password=pass
auth-url=https://<keystone_ip>/identity/v3
tenant-id=c869168a828847f39f7f06edd7305637
domain-id=2a73b8f597c04551a0fdc8e95544be8a

[metadata]
searchOrder = configDrive,metadata
```

Reply:
What I mean is, is `Metadata` already defined in `openstack_provider.Config`, isn't it?